### PR TITLE
Make text envelope types more consistent and improve where applicable

### PR DESCRIPTION
**Pull request by intricate**

### CODE REVIEW

File: cardano-api/src/Cardano/Api/Typed.hs [hunk 1]
```diff
@@ -1559,7 +1561,7 @@ instance HasTypeProxy Certificate where
     proxyToAsType _ = AsCertificate
 
 instance HasTextEnvelope Certificate where
-    textEnvelopeType _ = "Certificate"
+    textEnvelopeType _ = "CertificateShelley"
```
Comment (intricate) [inline cardano-api/src/Cardano/Api/Typed.hs:1564]: Should I suffix with `*Shelley` even though there's only one kind of certificate right now?

File: cardano-api/src/Cardano/Api/Typed.hs [hunk 2]
```diff
@@ -1957,7 +1959,7 @@ instance HasTypeProxy UpdateProposal where
     proxyToAsType _ = AsUpdateProposal
 
 instance HasTextEnvelope UpdateProposal where
-    textEnvelopeType _ = "Shelley update proposal"
+    textEnvelopeType _ = "UpdateProposalShelley"
```
Comment (intricate) [inline cardano-api/src/Cardano/Api/Typed.hs:1962]: Same question here.

File: cardano-api/src/Cardano/Api/Typed.hs [hunk 3]
```diff
@@ -3105,19 +3114,18 @@ instance SerialiseAsRawBytes (Hash PaymentKey) where
...
 
 instance HasTextEnvelope (SigningKey PaymentKey) where
-    textEnvelopeType _ = "SigningKeyShelley"
-    -- TODO: include the actual crypto algorithm name, to catch changes
-    -- TODO: fix these inconsistent names for the public testnet re-spin
+    textEnvelopeType _ = "PaymentSigningKeyShelley "
+                      <> fromString (Crypto.algorithmNameDSIGN proxy)
+      where
+        proxy :: Proxy (Shelley.DSIGN ShelleyCrypto)
+        proxy = Proxy
```
Comment (intricate) [inline cardano-api/src/Cardano/Api/Typed.hs:3128]: Ideally, I would have wanted to concat the strings without the `_` between them. However, some of the algorithm names are formatted differently to others (e.g. `ed25519`, `PraosVRF`), so I thought it'd look odd to see something like `PaymentSigningKeyShelleyed25519`.

File: cardano-api/src/Cardano/Api/Typed.hs [hunk 4]
```diff
@@ -3105,19 +3114,18 @@ instance SerialiseAsRawBytes (Hash PaymentKey) where
...
 
 instance HasTextEnvelope (SigningKey PaymentKey) where
-    textEnvelopeType _ = "SigningKeyShelley"
-    -- TODO: include the actual crypto algorithm name, to catch changes
-    -- TODO: fix these inconsistent names for the public testnet re-spin
+    textEnvelopeType _ = "PaymentSigningKeyShelley_"
+                      <> fromString (Crypto.algorithmNameDSIGN proxy)
+      where
+        proxy :: Proxy (Shelley.DSIGN ShelleyCrypto)
+        proxy = Proxy
```
Comment (intricate) [inline cardano-api/src/Cardano/Api/Typed.hs:3128]: Ideally, I would have wanted to concat the strings without the `_` between them. However, some of the algorithm names are formatted differently to others (e.g. `ed25519`, `PraosVRF`), so I thought it'd look odd to see something like `PaymentSigningKeyShelleyed25519`.

**Comment by dcoutts** [comment]
bors merge

**Comment by dcoutts** [comment]
bors merge

**Pull request merged**